architecture.md

```diff
@@ -0,0 +1,91 @@
+As an executable, it runs a simulation by launching two processes: the backend server process and frontend client process.
+
+The core functionality of Ignition Gazebo is the client-server communication and the various plugins it loads to facilitate the simulation.
+Plugins are libraries that are loaded at runtime.
+They allow Ignition Gazebo to utilize other Ignition libraries.
+There are many plugin types in the Ignition framework.
+For example, there are plugins that introduce new physics engines or rendering engines.
+However, for the purpose of this document, any mention of plugins is referring only to Ignition Gazebo server and GUI plugins.
+
+Because they’re loaded at runtime, Ignition does not need to be recompiled to add or remove plugins.
+Ignition Gazebo ships with many plugins by default, all of which are optional and can be removed by the user.
+Users can also add more plugins and even write their own plugins that will be compiled into library files.  
+
+Ignition libraries are modular.
+Plugins let Ignition Gazebo utilize other libraries.
+For example, [Ignition Physics](https://ignitionrobotics.org/libs/physics) and Ignition Gazebo are independent of one another, so Ignition Gazebo has a physics plugin that uses Ignition Physics as a library and incorporates Gazebo specifics, allowing Physics to be a system running in the simulation loop.
+
+The Ignition Physics library is only used in the physics plugin, not in other plugins nor in the core of Ignition Gazebo.
+Some libraries are only used by one plugin, or one in each process (frontend and backend).
+However, some foundational libraries are used by all plugins, like [Ignition Common](https://ignitionrobotics.org/libs/common) which provides common functionality like logging, manipulating strings, file system interaction, etc., to all plugins.
+Other such libraries include [Ignition Plugin](https://ignitionrobotics.org/libs/plugin), [Ignition Math](https://ignitionrobotics.org/libs/math), [SDFormat](https://ignitionrobotics.org/libs/sdformat), and more.
+
+There are certain plugins in both the frontend and backend processes that are loaded by default in every version of Ignition.
+Many other plugins are optional.
+One common optional plugin is the sensors plugin.
+Both optional and default plugins can be removed or added at any time; Ignition Gazebo will continue to run with limited functionality.
+These demos on [Server Configuration](https://ignitionrobotics.org/api/gazebo/4.3/server_config.html) and [GUI Configuration](https://ignitionrobotics.org/api/gazebo/4.3/gui_config.html) showcase that functionality.
+
+The simulation process is depicted in the diagram below, and further explained in the Backend and Frontend process sections that follow.
+
+![Ignition Gazebo architecture diagram](images/IgnitonGazeboArchicteture.png)
```

Comment:
This image is great! About time we had something like this! I love the little circles representing common Ignition libraries.

I just have some minor comments:

* The arrow that says "Packaged Messages" could say "Serialized ECM state" or something to that effect.
* How about using different arrows for the ones that represent Ignition Transport (the ones going across processes) and the ones connecting systems to the ECMs?
* Just a personal preference, but how about using a sans serif font? I think they're slightly cleaner. 

Finally, would it be possible for you to save this image as SVG and upload it as well? That should make it easier to edit it in the future. Thanks!